File: test/dev_mode_spec.dart
library angular2.test.dev_mode_spec;

import "package:angular2/testing_internal.dart"
    show
        AsyncTestCompleter,
        beforeEach,
        ddescribe,
        describe,
        expect,
        iit,
        inject,
        it,
        xdescribe,
        xit;
import "package:angular2/src/facade/lang.dart" show assertionsEnabled, IS_DART;

main() {
  describe("dev mode", () {
    it("is enabled in our tests by default", () {
      expect(assertionsEnabled()).toBe(true);
    });
  });
  if (IS_DART) {
    describe("checked mode", () {
      it("is enabled in our tests", () {
        try {
          String s = (42 as dynamic);
          expect(s).toEqual(42);
          throw "should not be reached";
        } catch (e, e_stack) {}
      });
    });
  }
}

File: modules/@angular/core/test/dev_mode_spec.ts
/**
 * @license
 * Copyright Google Inc. All Rights Reserved.
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://angular.io/license
 */

import {isDevMode} from '@angular/core';
import {beforeEach, ddescribe, describe, expect, iit, inject, it, xdescribe, xit} from '../testing';

export function main() {
  describe('dev mode', () => {
    it('is enabled in our tests by default', () => { expect(isDevMode()).toBe(true); });
  });
}
